app/controllers/products_controller.rb

```diff
@@ -0,0 +1,18 @@
+class ProductsController < ApplicationController
+  def index
+    @products = Product.all
+    @categories = Category.all
+  end
+  def show
+        @categories = Category.all
+        @product = Product.find(params[:id])
+        @comments = Comment.where(product_id:params[:id])
+  end
+  def new
+  end
+
+  def create
+    # ajout if logged in apres
+   Product.create!(name:params[:name],description:params[:description],image:params[:image])
```

Comment:
Line is too long. [92/80]<br>Use 2 (not 1) spaces for indentation.<br>Space missing after colon.<br>Space missing after comma.